src/commands/timeout.ts

```diff
@@ -54,14 +59,63 @@ export class TimeoutCommand extends Command {
       return;
     }
 
+    if (!reason) {
+      await interaction.reply({
+        content: `Error: Timeouts require a reason.`,
+        ephemeral: true,
+      });
+      return;
+    }
+
+    const filter = { '_id.guild': interaction.guildId, '_id.user': member.id };
+
+    const userTimeout: timeoutLog = {
+      date: new Date(),
+      duration: readableDuration,
+      user: interaction.user.id,
+      reason: reason
+    };
+
+    const update = {
+      $push: {
+        timeouts: userTimeout
+      }
+    };
+
+    const options = { upsert: true };
+
+    moderationLogs.findOneAndUpdate(filter, update, options);
+
     await member.timeout(durationMilliseconds, reason ?? undefined);
+    const expiration = new Date(interaction.createdTimestamp + durationMilliseconds);
+
+    const embed = new EmbedBuilder()
+      .setColor(Color.Red)
+      .setTitle('You Have Been Timed Out')
+      .addFields(
+        { name: 'Server', value: `${guild.name}` },
+        { name: 'Reason', value: reason },
+        { name: 'Duration', value: readableDuration },
+        {
+          name: 'Expiration',
+          value: time(expiration, TimestampStyles.RelativeTime),
+          inline: true,
+        },
+      )
+      .setTimestamp(interaction.createdTimestamp);
+
+    await member.send({ embeds: [embed] });
+
+    const ephemeralEmbed = new EmbedBuilder()
+      .setColor(Color.Red)
+      .setDescription(`${user.tag} timed out for ${readableDuration}`);
 
     await interaction.reply({
-      content: `${user.tag} timed out for ${readableDuration}`,
+      embeds: [ephemeralEmbed],
       ephemeral: true,
     });
 
-    await messageLogger.logMemberTimeout(
+    messageLogger.logMemberTimeout(
```

Comment:
Function is still async, so we should likely still await it, so that failures are captured properly.
```suggestion
    await messageLogger.logMemberTimeout(
```